### PR TITLE
[Incremental] Allow build record to reside at a relative path

### PR DESCRIPTION
**Pull request by davidungar**
Allow build record to reside at a relative path on the way to making legacy lit tests work.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
Part of rdar://71869992 (Swift Driver must handle relative paths in build records and other places in order to support legacy Driver/Dependencies tests)

File: Sources/SwiftDriver/Incremental Compilation/BuildRecordInfo.swift [hunk 1]
```diff
@@ -175,3 +178,12 @@ import SwiftOptions
     }
   }
 }
+
+extension VirtualPath {
```
Comment (artemcm) [inline Sources/SwiftDriver/Incremental Compilation/BuildRecordInfo.swift:182]: We already have:
```
extension VirtualPath {
  fileprivate func resolvedRelativePath(base: AbsolutePath) -> VirtualPath {
    guard case let .relative(relPath) = self else { return self }
    return .absolute(.init(base, relPath))
  }
}
```
In `OutputFileMap.swift`. Should we common these two out?

Comment (davidungar) [reply Sources/SwiftDriver/Incremental Compilation/BuildRecordInfo.swift:182]: Thanks for catching this! Will do!!

**Review by artemcm — APPROVED** [review]

**Comment by davidungar** [comment]
@swift-ci please test

**Pull request merged**